rqt_shell/src/rqt_shell/shell.py

```diff
@@ -85,9 +86,16 @@ def __init__(self, context):
         super(Shell, self).__init__(context)
         self._context = context
         self.setObjectName('Shell')
+        self.args = self._parse_args(context.argv())
 
         self._widget = None
 
+    def _parse_args(self, argv):
+        parser = argparse.ArgumentParser(prog='rqt_shell', add_help=False)
+        parser.add_argument('-i', '--init_script', default="",
```

Comment:
This is not how `argparse` works. Even with `--init-script` the argument will be accessible as `args.init_script`. (You can even use the `dest` argument to assign it to any variable name. Not necessary here, just fyi.)